nixos/modules/services/hardware/logiops.nix

```diff
@@ -0,0 +1,139 @@
+                    {
+                        type: "Gestures";
+                        gestures: (
+                            {
+                                direction: "Up";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type: "Keypress";
+                                    keys: ["KEY_UP"];
+                                };
+                            },
+                            {
+                                direction: "Down";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type: "Keypress";
+                                    keys: ["KEY_DOWN"];
+                                };
+                            },
+                            {
+                                direction: "Left";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type: "CycleDPI";
+                                    dpis: [400, 600, 800, 1000, 1200, 1400, 1600];
+                                };
+                            },
+                            {
+                                direction: "Right";
+                                mode: "OnRelease";
+                                action =
+                                {
+                                    type = "ToggleSmartshift";
+                                }
+                            },
+                            {
+                                direction: "None"
+                                mode: "NoPress"
+                            }
+                        );
+                    };
+                },
+                {
+                    cid: 0xc4;
+                    action =
+                    {
+                        type: "Keypress";
+                        keys: ["KEY_A"];
+                    };
+                }
+            );
+        }
+        );
+      '';
+    };
+
+    # settings = mkOption {
+    #   type = types.submodule {
+    #     # Declare that the settings option supports arbitrary format values, libconfig here
+    #     freeformType = settingsFormat.type;
+    #   };
+    #   default = {};
+    #   # Add upstream documentation to the settings description
+    #   description = ''
+    #     Configuration for `logiops`, see
+    #     <link xlink:href="https://github.com/PixlOne/logiops/wiki/Configuration"/>
+    #     for supported values.
+    #   '';
+    # };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.logiops = {
+      description = "Logitech Configuration Daemon";
+      startLimitIntervalSec = 0;
+      serviceConfig = {
+        Type = "simple";
+        ExecStart = "${pkgs.logiops}/bin/logid";
+        User = "root";
```

Comment:
Hi, I've also been tooling around with writing a service for logiops and I found out how to get around running the service as root. This is my config so far:

```
services.udev.extraRules = ''
  KERNEL=="uinput", GROUP="input", MODE:="0660", OPTIONS+="static_node=uinput"
  SUBSYSTEM=="hidraw", ATTRS{idVendor}=="046d", MODE="0660", GROUP="input"
'';
```

And in the service:

```
DynamicUser = true;
SupplementaryGroups = "input";
NoNewPrivileges = true;
ProtectSystem = "strict";
ProtectHome = true;
ProtectProc = "invisible";
ProcSubset = "pid";
PrivateTmp = true;
PrivateUsers = true;
PrivateMounts = true;
ProtectKernelTunables = true;
ProtectKernelModules = true;
ProtectKernelLogs = true;
ProtectControlGroups = true;
RestrictNamespaces = true;
```

Which seems to be working, as far as the Systemd restrictions go. Any thoughts?